src/dbnode/storage/index.go

```diff
@@ -1669,23 +1668,24 @@ func (i *nsIndex) queryWithSpan(
 			for !blockIter.iter.Done() {
 				// if this is not the first iteration of the iterator, need to acquire another permit.
 				if !first {
-					acq, waitTime := waitForPermit()
+					permit, waitTime = waitForPermit()
 					blockIter.waitTime += waitTime
-					if !acq {
+					if permit == nil {
 						break
 					}
 				}
 				first = false
 				startProcessing := time.Now()
-				execBlockFn(ctx, blockIter.block, blockIter.iter, opts, state, results, logFields)
+				execBlockFn(ctx, blockIter.block, permit, blockIter.iter, opts, state, results, logFields)
 				processingTime := time.Since(startProcessing)
 				queryMetrics.blockProcessingTime.RecordDuration(processingTime)
 				blockIter.processingTime += processingTime
-				permits.Release(int64(processingTime))
+				permit.Use(int64(processingTime))
```

Comment:
@ryanhall07 What's the motivation for calling .Use on the permit, if a line later it is immediately released, hence you're not using that information anywhere?

Reply:
so the `Permit` can track how much of the allowed quota was actually used. impls of the `Permits` may care to know about this information.